objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestYoloProcessor.java

```diff
@@ -131,6 +139,95 @@ public void testInvalidData() {
+        
+        // Only run test with intended vantiq availability
+        assumeTrue(testAuthToken != null && testVantiqServer != null);
+        labelTestHelper(false);
+        
+    }
+    
+    @Test
+    public void testImageSavingWithLabels() throws ImageProcessingException {
+        
+        // Only run test with intended vantiq availability
+        assumeTrue(testAuthToken != null && testVantiqServer != null);
+        labelTestHelper(true);
+        
+    }
+    
+    // Helper function to test the "labelImage" option
+    public void labelTestHelper(Boolean labelOption) throws ImageProcessingException {
+        Map config = new LinkedHashMap<>();
+        YoloProcessor ypImageSaver = new YoloProcessor();
+
+        config.put("pbFile", PB_FILE);
+        config.put("labelFile", LABEL_FILE);
+        config.put("outputDir", OUTPUT_DIR);
+        config.put("saveRate", SAVE_RATE);
+        config.put("saveImage", "local");
+        
+        if(labelOption) {
+            config.put("labelImage", "true");
+        } else {
+            config.put("labelImage", "false");
+        }
+        
+        try {
+            ypImageSaver.setupImageProcessing(config, MODEL_DIRECTORY, testAuthToken, testVantiqServer);
+        } catch (Exception e) {
+            fail("Could not setup the JSON YoloProcessor");
+        }
+        File d = new File(OUTPUT_DIR);
+        try {
+            // Ensure no results from previous tests
+            if (d.exists()) {
+                deleteDirectory(OUTPUT_DIR);
+            }
+
+            NeuralNetResults results = ypImageSaver.processImage(getTestImage());
+            assert results != null;
+            assert results.getResults() != null;
+            
+            // Should save first image with timestamp
+            assert d.exists();
+            assert d.isDirectory();
+            assert d.listFiles().length == 1;
+            assert d.listFiles()[0].getName().matches(timestampPattern);
+            
+            try {
+                // Converting original test image to buffered image, since this is how we save images
+                BufferedImage tempOriginal = ImageIO.read(new ByteArrayInputStream(getTestImage()));
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                ImageIO.write(tempOriginal, "jpg", baos);
+                baos.flush();
+                
+                // Getting both images as byte arrays and comparing them for equality
+                byte [] originalImg = baos.toByteArray();
+                byte [] savedImg = Files.readAllBytes(d.listFiles()[0].toPath());
+                
+                // Based on labelOption, check if they are equal or not equal
+                if (labelOption) {
+                    assertFalse("Images should not be identical", Arrays.equals(originalImg, savedImg));
+                } else {
+                    assert Arrays.equals(originalImg, savedImg);
```

Comment:
Would be better (& consistent, thus easier to read) if the "else assert" was an assertTrue with a description of the problem, analogous to the assertFalse above it.